core/trino-main/src/main/java/io/trino/type/LikeFunctions.java

```diff
@@ -76,6 +77,21 @@ public static LikeMatcher likePattern(@SqlType("varchar") Slice pattern, @SqlTyp
         }
     }
 
+    public static boolean isMatchAllPattern(Slice pattern)
+    {
+        if (!isAscii(pattern)) {
```

Comment:
This is unnecessary. The comparison below (`current != '%'`) ensures that the pattern contains only ASCII.